include/model.php

```diff
@@ -98,53 +108,26 @@ public function has_languages() {
 	 * @return array List of PLL_Language objects or PLL_Language object properties.
 	 */
 	public function get_languages_list( $args = array() ) {
-		if ( false === $languages = $this->cache->get( 'languages' ) ) {
-			$languages = array();
-
-			// Create the languages from taxonomies.
-			if ( ( defined( 'PLL_CACHE_LANGUAGES' ) && ! PLL_CACHE_LANGUAGES ) || false === ( $languages = get_transient( 'pll_languages_list' ) ) ) {
-				$languages = array();
-
-				$post_languages = $this->get_language_terms();
-
-				$term_languages = get_terms( array( 'taxonomy' => 'term_language', 'hide_empty' => false ) );
-				$term_languages = empty( $term_languages ) || is_wp_error( $term_languages ) ?
-					array() : array_combine( wp_list_pluck( $term_languages, 'slug' ), $term_languages );
+		$languages = $this->cache->get( 'languages' );
 
-				if ( ! empty( $post_languages ) && ! empty( $term_languages ) ) {
-					foreach ( $post_languages as $k => $v ) {
-						if ( isset( $term_languages[ 'pll_' . $v->slug ] ) ) {
-							$languages[ $k ] = new PLL_Language( $v, $term_languages[ 'pll_' . $v->slug ] );
-						}
+		if ( ! is_array( $languages ) ) {
+			if ( defined( 'PLL_CACHE_LANGUAGES' ) && ! PLL_CACHE_LANGUAGES ) {
+				// Create the languages from taxonomies.
+				$languages = $this->get_languages_from_taxonomies();
+			} else {
+				$languages = get_transient( 'pll_languages_list' );
+
+				if ( ! is_array( $languages ) ) {
+					// Create the languages from taxonomies.
+					$languages = $this->get_languages_from_taxonomies();
+				} else {
+					// Create the languages directly from arrays stored in the transient.
+					foreach ( $languages as $k => $v ) {
+						$languages[ $k ] = PLL_Language_Factory::create( $v );
```

Comment:
```suggestion
						$languages[ $k ] = PLL_Language_Factory::get( $v );
```

Reply:
Done in ebf0060f535ee436b809d597c99f6b8fc9dbdfa1